### ISSUE TITLE
[Master] Serverless Camp Feedback Improvement Ideas

### ISSUE THREAD

**Issue by lazyplatypus** [bitprj/Intro-To-Serverless#395]
Now that the Serverless Camp S21 is over, we want your feedback! What went good. What could be improved? Let's turn this into actionable next steps for the W22 Cohort.

**Comment by beatrixcendana** [comment]
What went good:
- Tools and plans were organized. I like how BitProject kept track of what students did with the Notion. 
- Having the weekly live stream is a good idea.
- I like how Emily explained the lessons so clearly. 

What should be improved:
- Asking the progress of students personally (how they are doing with projects, are they doing okay?). Not all students are comfortable asking questions during the work sessions. Or maybe there are some students who can't join the work sessions due to the timezone issue. 
- Asking the weekly progress of students (during the final project) by setting up quick check-in with instructors (it can be optional). If they don't have time, they have to submit a weekly progress video.
- Documentation should be more beginner-friendly. The step of the project should be clear what to do next. 
- I think we should also have fun sometimes, such as gathering with the mentors, instructors and playing games together (can do it weekly or bi-weekly at least). I mean it is not just having fun but also build networking. 
- I think it is also a good idea before starting the camp, students get training on how to use Git, Github (Maybe through workshops or talks.). So, there are no more questions regarding Github during the first camp. They can focus on learning the content of the camp. 
- Maybe BitProject can consider adding three more workshops during the camp ("How to create a great CFP", "How to be a great developer or software engineer, what practice/skills/habits that engineer needs to have", and "How to craft the presentation skills").

**Comment by a-padman** [comment]
During week 0 (or even earlier), provide students with the list of programs, accounts, and extensions they will need for the camp

**Comment by a-padman** [comment]
Add in content about debugging options like breakpoints, log points, or log messages in the output and log console in Azure to see deployed errors

**Comment by a-padman** [comment]
Include some short examples/explanations about how CSS works in the curriculums (or point to resources that have these)

**Comment by a-padman** [comment]
Continue providing students with Loom videos each week (it was especially useful for students when they saw Shreya debug her own errors/errors they might run in to)

**Comment by a-padman** [comment]
Work sessions are a bit disorganized when students are looking for help, maybe implement some sort of queue so they know when they can expect to be able to speak with an instructor

**Comment by a-padman** [comment]
If possible, add more explicit error messages from the counselor bot if possible (rather than 'no response'), as it would help students debug

**Comment by a-padman** [comment]
The Twilio documentation is hard to follow at first glance, add more explicit steps for this activity

**Comment by a-padman** [comment]
Many student-mentee pairs in general were either spending too much time on pair-programming/did not get a chance to complete it, any suggestions for how we can fix this?

**Comment by a-padman** [comment]
Feedback for the camp should **either be given on Github** through the counselor bot itself, **or with Polly** (I think having to give feedback in both places confused students, so the Polly responses rates teetered off by the end)

**Comment by creaturenex** [comment]
I would like to try and consolidate information or resources as much as possible. We were using a lot of sites, channels and sometimes looking for a link or video was kinda of annoying to keep track of. Also, Slack will get rid of older messages.

-Maybe a group calendly or share your calendar with your time block availability(you can keep them private) just to show when people are free especially the bit project team. For example when I was asking for help sometimes I would not get a response why? maybe they were in a meeting or working on something. For the students maybe they are in class at a certain time etc.

- Let me know the feasibility of this idea. Try to remove package/installation issues by either using an online environment like cloud 9 or installing a docker container with everything a student needs for the camp. This would be installed as a script so students do not have to go throw the trouble of installing docker etc themselves. Everything they do is save to the container and tested within the container.

**Comment by lazyplatypus** [comment]
From @ganning127 

Github Bot Changes
- Comment specific errors on the PR when things go wrong in the same thread
- There isn't one super long issue thread - it's one-to-one step to PR. 
- Cannot accidentally merge changes 
- Skip steps
- It's now serverless! on Azure Functions

Serverless Repo Curriculum Changes
- Made things more consistent
- rewrote tests so it saves the errors students make

**Comment by a-padman** [comment]
Improve communication by regularly check in with students weekly during final project implementation especially

**Comment by a-padman** [comment]
Camp Social Events: 
It was hard to find times everyone was available while the camp was in progress. I think there would maybe be a better turn out if X amount of socials were required? Is there a way we can better gauge student interest for events they'd like to see happen?

**Comment by lazyplatypus** [comment]
From @emsesc 
How do we not overwhelm counselors with question

Potential Fixes
- Find a platform to ticket complex issues
- Get more counselors
- Assign Counselors to particular students
